packages/pwa/app/utils/url.js

```diff
@@ -102,16 +103,24 @@ export const productUrlBuilder = (product) => encodeURI(`/product/${product.id}`
 export const searchUrlBuilder = (searchTerm) => `/search?q=${searchTerm}`
 
 /**
- * Returns a URL using the new locale.
+ * Returns a URL pathname using the new locale.
  *
- * @param previousLocale
- * @param newLocale
+ * @param {string} newLocale
+ * @param {Object} [options]
+ * @param {Object} options.location - original location (similar value to `window.location`)
  * @returns {string}
  */
-export const buildUrlLocale = (previousLocale, newLocale) => {
+export const buildUrlLocale = (newLocale, options = {}) => {
+    const location = options.location
+        ? options.location
+        : typeof window !== 'undefined'
+        ? window.location // to be used within event handlers
+        : useLocation() // to be used within React components
```

Comment:
We shouldn't be using a hook in a definition of a utility function, they should be use in 2 places, 1. in a component, 2. when creating another hook using composition.

Things have changed in `develop` so you might have to rethink this function, albeit the function in the `develop` branch also isn't the cleanest, so might want to think about changing the signature of the function to use dependancy injection.. so the sig might look more like `replaceLocaleInUrl(url, locale)` that way we don't have this branching for using window and useLocation, etc.

Reply:
You're right.. hook should not be used here in this case. All right, let me merge the latest changes from `develop` first.

Reply:
@bendvc Ok I've merged the latest `develop` and made some tweaks. Please take a look at the PR again. I think we can reuse the existing function signature. I piggy back on the `opts` object and allow it to accept `location`.